crates/autopilot/src/run_loop.rs

```diff
@@ -105,6 +105,8 @@ impl RunLoop {
             Class::Liquidity => true,
             Class::Limit => false,
         }) {
+            // Updating liveness probe to not report unhealthy due to this optimization
+            self.liveness.auction();
```

Comment:
Instead of adding this call twice in the code, can we lift it into the right place (before we decide that an auction is not worth syncing?)

This seems to be also needed for shadow...

Reply:
> before we decide that an auction is not worth syncing?)

Do you mean where we check if the old auction and the new auction were built on a different block. IMO it's not clear that this is the "right" place. We could have an issue where we don't see new blocks and built the auction on the same block over and over without noticing that the autopilot it in fact not settling any auctions. This would have also not addressed the issue with the skipped auctions.

I tried to avoid the second update call but it ended being more trouble than it's worth IMO.

> This seems to be also needed for shadow...

That's true. Will open another PR for the shadow code.

Reply:
I think there is (or should be a place) in the code where we are done with all the infra stuff (fetching block, loading data from db) and where our logic starts (do we even want to send the auction to solvers). In between we should make a single call to `liveness.auction` imo.